src/ca/mcgill/cs/jetuml/views/StringViewer.java

```diff
@@ -107,23 +145,36 @@ public Dimension getDimension(String pString)
 			return EMPTY;
 		}
 		Dimension dimension = getFontMetrics().getDimension(pString);
-		return new Dimension(Math.round(dimension.width() + HORIZONTAL_TEXT_PADDING*2), 
-				Math.round(dimension.height() + VERTICAL_TEXT_PADDING*2));
+		return new Dimension(Math.round(dimension.width() + aHorizontalPadding*2), 
+				Math.round(dimension.height() + aVerticalPadding*2));
 	}
 	
 	private TextAlignment getTextAlignment()
 	{		
-		if(aAlignment == Align.LEFT)
+		if ( aAlign == Alignment.TOP_LEFT || aAlign == Alignment.CENTER_LEFT || aAlign == Alignment.BOTTOM_LEFT )
 		{
 			return TextAlignment.LEFT;
 		}
-		else if(aAlignment == Align.CENTER)
+		else if ( aAlign == Alignment.TOP_CENTER || aAlign == Alignment.CENTER_CENTER || aAlign == Alignment.BOTTOM_CENTER )
 		{
 			return TextAlignment.CENTER;
 		}
 		return TextAlignment.RIGHT;
 	}
 	
+	private VPos getTextBaseline()
+	{
+		if ( aAlign == Alignment.TOP_LEFT || aAlign == Alignment.TOP_CENTER || aAlign == Alignment.TOP_RIGHT )
+		{

```

Comment:
You can define define methods on the enum for this, .e.g `isTop`, `isVerticallyCentered`, `isBottom`, etc.

Reply:
That makes sense; I added private methods to `Alignment` in the most recent commit